### PR TITLE
Better explain some client.rb settings

### PR DESCRIPTION
**Pull request by tas50**
In adding the chef_client_config resource I realized that some of these are not well spelled out and don't reflect what we want users adding.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: content/config_rb_client.md [hunk 1]
```diff
@@ -394,8 +393,11 @@ This configuration file has the following settings:
 
 `interval`
 
-:   The frequency (in seconds) at which Chef Infra Client runs. Default
-    value: `1800`.
+:   The frequency (in seconds) at which Chef Infra Client runs if running
+    Chef Infra Client in daemonized mode. It is recommended not to run
+    Chef Infra Client in daemonized mode and to instead rely on scheduled
+    Chef Infra Client execution from system schedulers like systemd timers,
+    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```
Comment (phiggins) [inline content/config_rb_client.md]:
```suggestion
:   The frequency (in seconds) at which Chef Infra Client runs when running
    in daemonized mode. It is recommended to not run
    in daemonized mode and to instead rely on scheduled
    execution from system schedulers like systemd timers,
    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```

This reads kind of awkwardly repeating "Chef Infra Client" so often.

File: content/config_rb_client.md [hunk 2]
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
 
 `minimal_ohai`
 
-:   Run the Ohai plugins for name detection and resource/provider
-    selection and no other Ohai plugins. Set to `true` during
-    integration testing to speed up test cycles.
+:   Run a minimal set of Ohai plugins providing data necessary for the
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming data sets such as
```
Comment (phiggins) [inline content/config_rb_client.md]:
```suggestion
    to true will skip many large and time consuming plugins such as
```

**Review by phiggins — APPROVED** [review]

File: content/config_rb_client.md [hunk 3]
```diff
@@ -394,8 +393,11 @@ This configuration file has the following settings:
 
 `interval`
 
-:   The frequency (in seconds) at which Chef Infra Client runs. Default
-    value: `1800`.
+:   The frequency (in seconds) at which Chef Infra Client runs when running
+    in daemonized mode. It is recommended to not run
+    in daemonized mode and to instead rely on scheduled
+    execution from system schedulers like systemd timers,
+    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```
Comment (IanMadd) [inline content/config_rb_client.md]:
```suggestion
    in daemonized mode. We do not recommend running
    in daemonized mode. Instead you should rely on scheduled
    execution from system schedulers like systemd timers,
    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```

File: content/config_rb_client.md [hunk 4]
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
 
 `minimal_ohai`
 
-:   Run the Ohai plugins for name detection and resource/provider
-    selection and no other Ohai plugins. Set to `true` during
-    integration testing to speed up test cycles.
+:   Run a minimal set of Ohai plugins providing data necessary for the
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming plugins such as
+    `cloud` or `packages`. Setting this this to true may break cookbooks
```
Comment (IanMadd) [inline content/config_rb_client.md]:
```suggestion
    `cloud` or `packages`. Setting this to true may break cookbooks
```

File: content/config_rb_client.md [hunk 5]
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
...
+    that assume all Ohai data will be present.
 
 `named_run_list`
 
-:   The run-list associated with a policy file.
+:   A specific named runlist defined in the node's applied Policyfile,
+    which the should be used when running Chef Infra Client.
```
Comment (IanMadd) [inline content/config_rb_client.md]:
```suggestion
:   A specific named runlist defined in the node's applied Policyfile
    which should be used when running Chef Infra Client.
```

File: content/config_rb_client.md [hunk 6]
```diff
@@ -464,11 +469,11 @@ This configuration file has the following settings:
 
 `node_name`
 
-:   The name of the node. Determines which configuration should be
+:   The name of the node. This determines which configuration should be
     applied and sets the `client_name`, which is the name used when
-    authenticating to a Chef Infra Server. The default value is the Chef
-    Infra Client FQDN, as detected by Ohai. In general, Chef recommends
-    that you leave this setting blank and let Ohai assign the FQDN of
+    authenticating to a Chef Infra Server. By default Chef Infra Client
```
Comment (IanMadd) [inline content/config_rb_client.md]:
```suggestion
    authenticating to a Chef Infra Server. By default, Chef Infra Client
```

File: content/config_rb_client.md [hunk 7]
```diff
@@ -430,9 +432,9 @@ This configuration file has the following settings:
 `log_level`
 
 :   The level of logging to be stored in a log file. Possible levels:
-    `:auto` (default), `:trace`(starting in 14.0.202), `:debug`, `:info`, `:warn`, `:error`,
-    or `:fatal`. Default value: `:warn` (when a terminal is available)
-    or `:info` (when a terminal is not available).
+    `:auto` (default), `:trace`, `:debug`, `:info`, `:warn`, `:error`,
+    or `:fatal`. The `:auto` level will use `:warn` when a terminal is
+    available or `:info` when a terminal is not available.
```
Comment (lamont-granquist) [inline content/config_rb_client.md:437]: This may be wrong now, I think we use `:info` by default and `:warn` is only used now when switching to syslog-ing, or when the user explicitly does `--force-logger`.

Comment (lamont-granquist) [reply content/config_rb_client.md:437]: (honestly I'm not sure how to document this right now for Chef-16, that's the best I can come up, I would need to burn a lot of time testing it, and I wouldn't know how to English that up for user consumption since it is a pile of edge conditions nearly nobody should care about any more)

Comment (tas50) [reply content/config_rb_client.md:437]: @lamont-granquist Wanna toss a suggestion to change that?

File: content/config_rb_client.md [hunk 8]
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
...
 `minimal_ohai`
 
-:   Run the Ohai plugins for name detection and resource/provider
-    selection and no other Ohai plugins. Set to `true` during
-    integration testing to speed up test cycles.
+:   Run a minimal set of Ohai plugins providing data necessary for the
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming plugins such as
+    `cloud` or `packages`. Setting this to true may break cookbooks
+    that assume all Ohai data will be present.
```
Comment (lamont-granquist) [inline content/config_rb_client.md:453]: Runs only the Ohai plugins that determine the O/S platform, hostname, and init system.  This can be used to speed up testing or `chef-apply` scripts, but will break many cookbooks which expect more extended Ohai attributes.

File: content/config_rb_client.md [hunk 9]
```diff
@@ -464,11 +469,11 @@ This configuration file has the following settings:
...
-:   The name of the node. Determines which configuration should be
+:   The name of the node. This determines which configuration should be
     applied and sets the `client_name`, which is the name used when
-    authenticating to a Chef Infra Server. The default value is the Chef
-    Infra Client FQDN, as detected by Ohai. In general, Chef recommends
-    that you leave this setting blank and let Ohai assign the FQDN of
+    authenticating to a Chef Infra Server. By default, Chef Infra Client
+    will use the system's FQDN as the node name. In general, Chef recommends
+    that you leave this setting blank and let the client assign the FQDN of
     the node as the `node_name` during each Chef Infra Client run.
```
Comment (lamont-granquist) [inline content/config_rb_client.md:477]: Feels like there should be something here about the "unique identity" of the node.

Comment (tas50) [reply content/config_rb_client.md:477]: Fixed everywhere

File: content/config_rb_client.md (hunk 8 above)
Comment (tas50) [reply content/config_rb_client.md:453]: That's all there. People don't need to know about chef testing here.

File: content/config_rb_client.md (hunk 7 above)
Comment (tas50) [reply content/config_rb_client.md:437]: From Slack "Just ship it"

**Review by IanMadd — APPROVED** [review]
